### PR TITLE
Unbreak haskellPackages.hpack-dhall

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:143]: Is this something you could report upstream?

If so, please create an issue upstream, and link to it here in this commit right above the `hpack-dhall = ` line.

We like to be able to have some way of tracking when these uses of `dontCheck` can be removed.

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1481]: This override belongs in the configuration-nix.nix

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1501]: I guess this will go away, when you rebase on haskell-updates.

**Review by maralorn — CHANGES_REQUESTED** [review]
I guess, this PR is pretty uncontroversial, but I can‘t really judge it before a rebase.

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (KAction) [reply pkgs/development/haskell-modules/configuration-common.nix:143]: It is already in master, but not in release: https://github.com/cabalism/hpack-dhall/commit/5049aa4dd9dc79b41d11e2c72dffddd9bc92b203

**Comment by KAction** [comment]
@maralorn @cdepillabout Rebases, fixed merge conflicts, found at which commit upstream already fixed problem. We will be able to drop override on next upstream release. Should I create issue about it?

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (maralorn) [reply pkgs/development/haskell-modules/configuration-common.nix:143]: Suggestion: Can you add an assert on hpack-dhall.version with a comment to remove the dontCheck, when the assert fails?

**Review by maralorn — CHANGES_REQUESTED** [review]

**Comment by maralorn** [comment]
Beautiful!

@GrahamcOfBorg build haskellPackages.hpack-dhall

Ready to merge, when test passes.

**Review by maralorn — APPROVED** [review]

**Comment by KAction** [comment]
@maralorn ofBorg seems to be unhappy, but it complains about pandoc, which seems to be unrelated.

**Comment by maralorn** [comment]
Yeah, someone should fix that … I meant to do that the last two days …

Anyways I'll quickly do the build locally.

**Comment by maralorn** [comment]
Checked build locally.

**Pull request merged**
